src/backend/crud/conversation.py

```diff
@@ -37,13 +37,21 @@ def get_conversation(
     """
     return (
         db.query(Conversation)
-        .filter(Conversation.id == conversation_id, Conversation.user_id == user_id)
+        .filter(
```

Comment:
Can we add the conditional filter here instead? You can chain .filter() operations 

eg:
```
query = <the previous query>
if agent_id is not None:
      query = query.filter(Conversation.agent_id == agent_id)

return query.first()
```